File: app/src/main/java/org/scp/gymlog/ui/common/animations/ResizeWidthAnimation.java
package org.scp.gymlog.ui.common.animations;

import static org.scp.gymlog.util.FormatUtils.toDp;

import android.view.animation.Transformation;
import android.view.View;
import android.view.animation.Animation;

public class ResizeWidthAnimation extends Animation {
    private final int mWidth;
    private final int mStartWidth;
    private final View mView;

    public ResizeWidthAnimation(View view, int width, long duration) {
        mView = view;
        mWidth = toDp(view.getResources().getDisplayMetrics(), width);
        mStartWidth = view.getWidth();
        setDuration(duration);
    }

    @Override
    protected void applyTransformation(float interpolatedTime, Transformation t) {
        int newWidth = mStartWidth + (int) ((mWidth - mStartWidth) * interpolatedTime);
        mView.getLayoutParams().width = newWidth;
        mView.requestLayout();
    }

    @Override
    public void initialize(int width, int height, int parentWidth, int parentHeight) {
        super.initialize(width, height, parentWidth, parentHeight);
    }

    @Override
    public boolean willChangeBounds() {
        return true;
    }
}

File: app/src/main/java/org/scp/gymlog/ui/common/animations/ResizeWidthAnimation.kt
package org.scp.gymlog.ui.common.animations

import android.view.View
import android.view.animation.Animation
import android.view.animation.Transformation
import org.scp.gymlog.util.FormatUtils

class ResizeWidthAnimation(
    private val mView: View,
    width: Int,
    duration: Long,
    toDp: Boolean = false
) : Animation() {

    private val mStartWidth: Int = mView.width
    private val mWidth: Int = if (toDp)
        FormatUtils.toDp(mView.resources.displayMetrics, width) else
        width

    init {
        setDuration(duration)
    }

    override fun applyTransformation(interpolatedTime: Float, t: Transformation) {
        val newWidth = mStartWidth + ((mWidth - mStartWidth) * interpolatedTime).toInt()
        mView.layoutParams.width = newWidth
        mView.requestLayout()
    }

    override fun willChangeBounds(): Boolean {
        return true
    }
}